### PR TITLE
Some compaction code cleanup

### PR DESCRIPTION
**Pull request by milleruntime**

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java [hunk 1]
```diff
@@ -754,14 +754,9 @@ public void run() {
...
-
-    this.compactionManager = new CompactionManager(new Iterable<Compactable>() {
-      @Override
-      public Iterator<Compactable> iterator() {
-        return Iterators.transform(onlineTablets.snapshot().values().iterator(),
-            Tablet::asCompactable);
-      }
-    }, getContext(), ceMetrics);
+    var tabletIterator = onlineTablets.snapshot().values().iterator();
+    Iterable<Compactable> compactables = () -> transform(tabletIterator, Tablet::asCompactable);
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java:758]: This changes the behavior from getting a snapshot each time the iterator is requested to only getting a snapshot once ever.  Also if the iterator over the snapshot is only created once, then once its exhausted its seem like this iterable will always return an empty iterator.  I think the previous behavior needs to be maintained and the following might do it.

```suggestion
    Iterable<Compactable> compactables = () -> transform( onlineTablets.snapshot().values().iterator(), Tablet::asCompactable);
```

Comment (milleruntime) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java:758]: Good catch. I don't see how yours is any different though.

**Comment by ctubbsii** [comment]
@milleruntime Was this closed because it was superseded by a different change, or because it's not needed, or something else?

**Comment by milleruntime** [comment]
> @milleruntime Was this closed because it was superseded by a different change, or because it's not needed, or something else?

It's not needed. And the uncertainty of what is going on with that Lambda.

**Pull request closed without merging**